src/cider/tests/utils/CiderArrowChecker.cpp

```diff
@@ -43,27 +43,81 @@ bool checkArrowBuffer(const struct ArrowArray* expect_array,
 
   auto expect_null_buffer = reinterpret_cast<const uint8_t*>(expect_array->buffers[0]);
   auto actual_null_buffer = reinterpret_cast<const uint8_t*>(actual_array->buffers[0]);
-  if (expect_null_buffer == nullptr && actual_null_buffer == nullptr) {
+
+  if (expect_null_buffer && actual_null_buffer) {
```

Comment:
is it possible to detect a scenario and pop message like "values are equal but nulls are different"?